federation/pkg/federatedtypes/adapter.go

```diff
@@ -56,6 +56,14 @@ type FederatedTypeAdapter interface {
 	NewTestObject(namespace string) pkgruntime.Object
 }
 
+// SchedulerAdapter defines additional operations for those controllers which need
+// additional scheduling logic.
+type SchedulerAdapter interface {
```

Comment:
I think the convention in go is to limit the scope of interfaces, though.  Also, YAGNI.  If/when a need arises the decision to avoid embedding can be revisited.